lib/plugins/aws/package/compile/events/alb/lib/targetGroups.js

```diff
@@ -16,6 +27,20 @@ module.exports = {
         functionName
       );
 
+      const healthCheckProperties = { HealthCheckEnabled: healthCheckDefaults.HealthCheckEnabled };
+      if (healthCheck) {
+        healthCheckProperties.HealthCheckEnabled = healthCheck.enabled;
+        healthCheckProperties.HealthCheckIntervalSeconds = healthCheck.intervalSeconds;
+        healthCheckProperties.HealthCheckPath = healthCheck.path;
+        healthCheckProperties.HealthCheckTimeoutSeconds = healthCheck.timeoutSeconds;
+        healthCheckProperties.HealthyThresholdCount = healthCheck.healthyThresholdCount;
+        healthCheckProperties.UnhealthyThresholdCount = healthCheck.unhealthyThresholdCount;
+        if (healthCheck.matcher && healthCheck.matcher.httpCode) {
+          healthCheckProperties.Matcher = { HttpCode: healthCheck.matcher.httpCode };
+        }
+        _.defaults(healthCheckProperties, healthCheckDefaults);
```

Comment:
`_.defaults` is not great as it won't override `null` and I believe in such case we should also jump in with default.

Therefore safest is probably to go as follows:

```javascript
Object.assign(healthCheckProperties, healthCheckDefaults);

if (healthCheck.enabled != null) healthCheckProperties.HealthCheckEnabled = healthCheck.enabled;
if (healthCheck.intervalSeconds != null)  healthCheckProperties.HealthCheckIntervalSeconds = healthCheck.intervalSeconds;
// etc.
```
In a future we would be able to express it more neatly with [logical nullish assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_nullish_assignment)